test/integration/infrastructure/infrastructure_test.go

```diff
@@ -489,6 +491,94 @@ var _ = Describe("Infrastructure tests", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	Context("with invalid credentials", func() {
+		It("should fail creation but succeed deletion", func() {
+			namespaceName, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+
+			providerConfig := newInfrastructureConfig(nil, nil, nil, false)
+
+			var (
+				namespace *corev1.Namespace
+				cluster   *extensionsv1alpha1.Cluster
+				infra     *extensionsv1alpha1.Infrastructure
+			)
+
+			framework.AddCleanupAction(func() {
+				By("cleaning up namespace and cluster")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+				Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
```

Comment:
In the other tests, we have these in the defer part. Can I ask the reason to add these here instead ?

Reply:
The code I refer to https://github.com/gardener/gardener-extension-provider-aws/pull/325/files#diff-159fa2b5244aa8582f3ab173f88010e046ae979b06ef94e88b0346b1d308dc50R272-R278.  It seems we don't want to leak any resources even if defer not called.

Reply:
@kon-angelo